src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -187,5 +190,102 @@ public void SetAccessControl_FileStream_FileSecurity_Success()
+            DirectoryInfo info = new DirectoryInfo("path");
+            Assert.Throws<ArgumentNullException>(() => info.Create(null));
+        }
+
+        [Fact]
+        public void DirectoryInfo_Create_NotFound()
+        {
+            DirectoryInfo info = new DirectoryInfo(@"W:\\I\\Do\\Not\\Exist");
+            DirectorySecurity security = new DirectorySecurity();
+            Assert.Throws<DirectoryNotFoundException>(() => info.Create(security));
+        }
+
+        [Fact]
+        public void DirectoryInfo_Create_AccessDenied()
+        {
+            DirectoryInfo info = new DirectoryInfo(@"C:\\Windows\\System32\\deniedaccess");
+            DirectorySecurity security = new DirectorySecurity();
+            Assert.Throws<UnauthorizedAccessException>(() => info.Create(security));
+        }
+
+        [Fact]
+        public void DirectoryInfo_Create_DefaultDirectorySecurity()
+        {
+            DirectorySecurity security = new DirectorySecurity();
+            CreateAndVerifyDirectoryWithSecurity(security);
+        }
+
+        [Theory]
+        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.FullControl, AccessControlType.Allow)]
+        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.ReadAndExecute, AccessControlType.Allow)]
+        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.Write, AccessControlType.Allow)]
+        public void DirectoryInfo_Create_DirectorySecurityWithSpecificAccessRule(WellKnownSidType sid, FileSystemRights rights, AccessControlType controlType)
+        {
+            DirectorySecurity security = new DirectorySecurity();
+            SecurityIdentifier identity = new SecurityIdentifier(sid, null);
+            FileSystemAccessRule accessRule = new FileSystemAccessRule(identity, rights, controlType);
+
+            security.AddAccessRule(accessRule);
+
+            CreateAndVerifyDirectoryWithSecurity(security);
+        }
+
+        private void CreateAndVerifyDirectoryWithSecurity(DirectorySecurity security)
+        {
+            using var directory = new TempDirectory();
+
+            string path = Path.Combine(directory.Path, "directory");
+            DirectoryInfo info = new DirectoryInfo(path);
+
+            info.Create(security);
+
+            Assert.True(Directory.Exists(path));
+            Assert.Equal(typeof(FileSystemRights), security.AccessRightType);
+
+            DirectorySecurity actualSecurity = info.GetAccessControl();
+            Assert.Equal(typeof(FileSystemRights), actualSecurity.AccessRightType);
+
+            List<FileSystemAccessRule> expectedRules = security.GetAccessRules(includeExplicit: true, includeInherited: false, typeof(SecurityIdentifier))
+                .Cast<FileSystemAccessRule>().ToList();
+
+            List<FileSystemAccessRule> actualRules = actualSecurity.GetAccessRules(includeExplicit: true, includeInherited: false, typeof(SecurityIdentifier))
+                .Cast<FileSystemAccessRule>().ToList();
+
```

Comment:
What about `GetAuditRules` -  should we also pull those (with `includeInherited: false`) and check there (presumably) aren't any?

Reply:
Looks like you would have to do `security.AddAuditRule` in the test as well. I guess at this point we are really testing `DirectorySecurity` and its friends rather than your code.

Reply:
Which is not a terribly bad thing, up to you

Reply:
I could add those tests too. @JeremyKuhne what do you think? Is it enough with what I have or do we want to be super paranoid?